### DISCUSSION TITLE
Dynamic Icon in Icon Widget and Icon Box Widget

### DISCUSSION THREAD

**Discussion by awakekat** [elementor/elementor#21743] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

I would like to make the icon in both the Icon Widget and the Icon Box Widget to be dynamic so that I can link a field with ACF and be able to bind an icon associated with a post and custom field. The link and title are dynamic but not the icon.

### Describe the solution you'd like

I would like to be able to link a custom field to the icon so that when I create a post, the icon is chosen for that specific post, category or tag. There is no dynamic choice with the icon.

### Describe alternatives you've considered

I could possibly dynamically bind a different icon with Unlimited Elements but I really do not want to have to use another plugin and go to the effort of creating a custom widget when all that need to be done in Elementor is making the icon in the Icon Widget and Icon Box Widget dynamic so that I can bind it easily to an ACF custom field.

**Comment by thorsolutions** [comment]
I am not sure why it makes sense to have dynamic "Link" but not the icon itself? Hopefully this is resolved soon. Ideally simply give it the same abilities like the image widget for dynamic content. 

![image](https://github.com/elementor/elementor/assets/16325700/7db0bdaa-1eb7-453d-a3eb-3d72598030ed)

**Comment by briski-design** [comment]
+1

**Comment by TimSgr** [comment]
+1

**Comment by dsusoftech** [comment]
Yes required feature

**Comment by projectm07** [comment]
Absolutely needed

**Comment by WajdiR** [comment]
+1 for this feature

**Comment by steveindzine** [comment]
+1

**Comment by CordialGit** [comment]
This is truly a needed feature.

You can see me facing the problem here-- https://wordpress.org/support/topic/cant-display-on-frontend/#new-topic-0

Please Elementor, add Dynamic link to Icons in Icon, Icon Box and Icon List Widgets.

**Comment by tjomahawk** [comment]
If i wouldn't see this myself i wouldn't believe it. This is a really basic feature.

**Comment by XeniaPro** [comment]
+ 1

**Comment by ofmarconi** [comment]
+1

**Comment by Nachtkastje** [comment]
+1

**Comment by briski-design** [comment]
+1

**Comment by aznalfl** [comment]
I've created a basic Dynamic Icon widget that anyone can download and use. https://github.com/aznalfl/elementor-dynamic-icon-widget

**Reply by admazy** [reply]
Thanks for a great plugin! Is there a way to make it work with https://wordpress.org/plugins/advanced-custom-fields-font-awesome/ ?

**Reply by iamabhishekgarg** [reply]
@aznalfl  its giving error in mine

**Comment by lioncatde** [comment]
+1

**Comment by nadworks** [comment]
+1
How's this not done already?

**Comment by pixelwisestudio** [comment]
This plugins solve the problem: https://acfelementorcustomicons.com/

**Comment by socenpauriba** [comment]
+1

**Comment by NN01010101** [comment]
+1

**Comment by thehervy** [comment]
+1

**Comment by AF1QUE** [comment]
+1

**Comment by mattsandler** [comment]
+1

**Comment by Kocarik** [comment]
+1

**Comment by eisenpunk** [comment]
+1

**Comment by wikiredargentina** [comment]
+1 please!!!

**Comment by Tokyaina** [comment]
+1
